File: Solutions/102-Binary-Tree-Level-Order-Traversal/Solution 2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        // Breadth-first Search
        // Recursive solution
        vector<vector<int>> ret;
        
        levelOrder(root, ret, 1);
        
        return ret;
    }
private:
    void levelOrder(TreeNode* root, vector<vector<int>>& ret, int level){
        if(root == NULL) return;
        if(ret.size() == level-1){
            vector<int> lev;
            lev.push_back(root->val);
            ret.push_back(lev);
        }else{
            ret[level-1].push_back(root->val);
        }
        
        levelOrder(root->left, ret, level+1);
        levelOrder(root->right, ret, level+1);
    }
};

File: Solutions/094-Binary-Tree-Inorder-Traversal/Solution2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        ret = []
        if not root:
            return []
        left = self.inorderTraversal(root.left)
        right = self.inorderTraversal(root.right)
        if left:
            ret.extend(left)
        ret.append(root.val)
        if right:
            ret.extend(right)
            
        return ret
